### PR TITLE
WT-10786 Use unique object prefixes in C tiered storage tests

### PR DESCRIPTION
**Pull request by pmacko86**

### CODE REVIEW

File: test/utility/tiered.c
Comment (ddanderson) [inline test/utility/tiered.c:217]: If we use embedded '/' for the prefix, I wonder how this will work (or not?) with dir_store?  Clearly dirstore could create the needed subdirectories in the bucket directory, but I don't see anything like that here.  Am I missing something?

**Review by ddanderson — COMMENTED** [review]
I have a question before I approve.  Otherwise LGTM (modulo indentation nit).

File: test/utility/tiered.c
Comment (pmacko86) [reply test/utility/tiered.c:214]: Yes, that's a bug. I forgot to run "s_all," which would fix it 😓...

File: test/utility/tiered.c
Comment (pmacko86) [reply test/utility/tiered.c:217]: The implementation of `testutil_tiered_storage_configuration` currently uses the old, default prefix `pfx-` for dir_store:
```
        testutil_check(__wt_snprintf(tiered_cfg, tiered_cfg_size, TESTUTIL_ENV_CONFIG_TIERED,
          is_dir_store ? dir : s3_bucket_name, is_dir_store ? "pfx-" : s3_prefix,
          opts->local_retention, opts->tiered_storage_source, auth_token));
```
Would it be okay to keep it this way, or should we add the path encoding also to dir_store?

Note that the use of the `'/'` character is currently mandated by our S3 lifecycle management policy.

**Review by ddanderson — APPROVED** [review]
Thanks for the answer!  Approved.

File: test/utility/util_random.c
Comment (keitharnoldsmith) [inline test/utility/util_random.c:43]: (very) minor nit: The session argument _is_ used, if not NULL, to ensure session's have monotonically increasing clocks. So maybe say that the session arg "is not required" or that the function "accepts a NULL session"?

**Review by keitharnoldsmith — APPROVED** [review]
LGTM -- One small nitpick about a comment.

**Pull request merged**
